### ISSUE TITLE
Warning while compiling hpx

### ISSUE THREAD

**Issue by justwagle** [STEllAR-GROUP/hpx#2109]
While compiling the latest master branch of hpx, invalid attribute warnings are shown. 
The full log of the compilation is linked below. 

https://gist.github.com/justwagle/c1a16b6d2dc5bacc8924396bb96bf562

**Comment by K-ballo** [comment]
This is a side-effect of the deprecation of `hpx_fwd.hpp`. Presumably those forward declarations mentioned in the warnings are seen by the compiler *after* the class was already defined (with the correct export attribute).

**Comment by W4RH4WK** [comment]
I encountered a similar issue recently where the `thread` class is already defined.

Output:
<https://gist.github.com/W4RH4WK/abf89baf6e03a9c0ba0f6cba170680f9>

Minimum Example:
```cpp
#include <hpx/include/serialization.hpp>
#include <hpx/runtime/serialization/list.hpp>
#include <hpx/include/threads.hpp>

#include <iostream>

int main(void) {
        std::cout << "Hello World" << std::endl;
        return 0;
}
```

Currently on:
b603449e6b97ced17adb829b07413f41b2ab6131

**Comment by hkaiser** [comment]
@W4RH4WK Thanks for this report. This should be fixed by merging #2429.
